neqo-transport/src/connection/mod.rs

```diff
@@ -225,6 +225,12 @@ impl AddressValidationInfo {
     }
 }
 
+struct ConnectionParameters {
+    pub quic_version: QuicVersion,
+    pub cc_algorithm: CongestionControlAlgorithm,
+    pub max_streams: Option<u64>,
+}
```

Comment:
I think that you might want to double-down on this model more heavily by making this struct public.  However, I would not expose the member fields or we get worse compatibility problems as we upgrade.  I know that having a bunch of getters and setters is annoying, but I think that is the right strategy here.

Using setters allows us to have defaults attached to this struct.  You can use the builder pattern with functions that use the `fn max_streams_remote_bidi(self, v: u64) -> Self` signature to make using this easy.  Then you don't need to manage `Option<T>` instances everywhere.

Also, can you have this implement `Clone`.  The server class will need that if we make it a public interface.

Also, can you split the stream limit up by type?  (We'll need to do the same for flow control, but that can come later.)